quantumblacklabs/kedro-viz
# KED-2875: Set up routing infrastructure and component placeholders for experiment tracking page

## Description

related ticket: [KED-2875](https://jira.quantumblack.com/browse/KED-2875)

This PR is to implement the initial routing setup as outlined [here](https://miro.com/app/board/o9J_lvLE7Ak=/) as part of the spike work for implement routing for Kedro-viz. Given the need to test the result of the routes, I had also set up simple components as placeholders according to the designs of the experiment tracking screenshots[here](https://mckinsey.invisionapp.com/share/RV15L5YZAP#/screens/164218949?browse) in order to effectively illustrate the effects of the routing setup. 

## Development notes

I had implemented the following: 

- the flowchart wrapper to contain all components to be displayed under the main flowchart page, and moved all related components from wrapper to this new component.
- the experiment wrapper to contain all components relating to the experiment tracking page
- The experiment tracking folder that will contain all components related to the experiment tracking feature
- Sidebar component for runsList, which includes the collapsable panel, as well as placeholder wording to indicate the routing 
- Details  componentfor runsList, which includes the styling to cater for the collapsable panel, as well as placeholder wording to indicate the routing 
- Routing setup in Wrapper.js for `/runsList`
- Routing setup in GlobalSidebar
- Routing setup in both sidebar and details component to process query params in the URL for the runIds / compareIds
- updates to the GlobalSidebar test now that we need to include the `MemoryRouter` in order for the component to render in tests

I have also:
- deleted the unneeded `visible` prop in the Globalsidebar & relevant tests given that is's no longer needed. 

Note: 

- I had not added tests for this PR given this acts more as a WIP in kickstarting the structure and routing setup for further development, where more functional tests can be added once we progress along in building out the actual feature
-  Given this is a WIP that will only be merged into a feature branch as part of the collaboration in building the FE work, I am only requesting @tynandebold review given the context in our previous discussion ( of course, all comments are still welcomed)
- While setting up the routes I discovered a problem with the requesting injecting the route into the URL for the request (i.e using `/runs` in the URL will result in the app calling `/runs/api/main`), however, on cross-checking multiple parts of the data ingestion setup within the app, I can confirm that the URL injected by the FE is correct. This issue, along with any refactor required to simplify to current data ingestion layer, will be dealt with in a separate ticket. 

## QA notes

With the error as mentioned in the note above, currently the routing on the deployed app would not work until the issue is resolved. In the meantime, please QA via running locally. 

- when running the local app, check if clicking the flowchart bring you to the main page, and clicking on the 'experiment' button brings you to `/runsList`. You should see the following:

![image](https://user-images.githubusercontent.com/27775658/136208045-bef41ccb-0301-4ba3-82e4-973af4738b58.png)

- try re-loading  the page via this URL`http://localhost:4141/runsList?run=1234` - you should see the runsID on both the sidebar and the details section like the following:

![image](https://user-images.githubusercontent.com/27775658/136208254-579058dc-9864-4ae5-81ab-45fa7426c5fb.png)

- try re-loading  the page via this URL`http://localhost:4141/runsList?compare=1234+2345+3456` - you should see the runsID on both the sidebar and the details section like the following:

![image](https://user-images.githubusercontent.com/27775658/136210295-7f77fefc-09f3-4eee-824a-1474fb1dbfa5.png)

## Checklist


## Legal notice


- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
